x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/planner/ExpressionTranslators.java

```diff
@@ -324,6 +324,41 @@ static Query translate(BinaryComparison bc, TranslatorHandler handler) {
             if (DataTypes.isDateTime(attribute.dataType())) {
                 zoneId = bc.zoneId();
             }
+            Query translated = expressionDispatcher.apply(bc, attribute, source, name, value, format, isDateLiteralComparison, zoneId);
+            if (translated != null) {
+                return translated;
+            }
+
+            throw new QlIllegalArgumentException("Don't know how to translate binary comparison [{}] in [{}]", bc.right().nodeString(), bc);
+        }
+
+        /**
+         * Represents a function which maps between an expression type and a query type, in a given translation context.
+         */
+        @FunctionalInterface
+        public interface Dispatcher {
```

Comment:
Probably want a better name for this, but I didn't want to spend a lot of time thinking about it.